### PR TITLE
Fog dependency

### PR DESCRIPTION
**Pull request by dhalai**
Hi,
You check file type with `CarrierWave::Storage::Fog::File` constant, but without installed `fog` it's undefined. At the same time, `fog` isn't required explicitly, so we can just check it.

### CODE REVIEW

**Comment by PikachuEXE** [comment]
Is calling `defined?` or requiring it and check the object's class everytime faster?

**Comment by dhalai** [comment]
It doesn't matter, the main reason is that `fog` has a lot of dependencies and it isn't necessary to have it here.

File: lib/carrierwave/bombshelter.rb [hunk 1]
```diff
@@ -59,14 +59,19 @@ def check_pixel_dimensions!(image)
...
       when CarrierWave::SanitizedFile, CarrierWave::Uploader::Base
         get_file(file.file)
       else
         file
       end
     end
+
+    def in_fog?(file)
+      defined?(CarrierWave::Storage::Fog) &&
+        file.instance_of?(CarrierWave::Storage::Fog::File)
```
Comment (DarthSim) [inline lib/carrierwave/bombshelter.rb]: Use `is_a?` here

File: lib/carrierwave/bombshelter.rb [hunk 2]
```diff
@@ -59,14 +59,19 @@ def check_pixel_dimensions!(image)
     end
 
     def get_file(file)
+      return file.url if in_fog?(file)
+
       case file
```
Comment (DarthSim) [inline lib/carrierwave/bombshelter.rb]: `case` isn't needed anymore. Use guard clause here.

**Comment by DarthSim** [comment]
My bad. Please take into account the comments and I'll be happy to merge this.

**Comment by dhalai** [comment]
thx, completed

File: lib/carrierwave/bombshelter.rb [hunk 3]
```diff
@@ -59,14 +59,24 @@ def check_pixel_dimensions!(image)
...
+      file
+    end
+
+    def in_fog?(file)
+      defined?(CarrierWave::Storage::Fog) &&
+        file.is_a?(CarrierWave::Storage::Fog::File)
+    end
+
+    def is_wrapped?(file)
+      file_wrappers.reduce(false) do |acc, wrapper|
```
Comment (DarthSim) [inline lib/carrierwave/bombshelter.rb]: It's better to use [`any?`](http://ruby-doc.org/core-2.3.0/Array.html#method-i-any-3F) here

File: lib/carrierwave/bombshelter.rb [hunk 4]
```diff
@@ -59,14 +59,24 @@ def check_pixel_dimensions!(image)
...
+        acc || file.is_a?(wrapper)
       end
     end
+
+    def file_wrappers
+      [CarrierWave::SanitizedFile, CarrierWave::Uploader::Base]
+    end
```
Comment (DarthSim) [inline lib/carrierwave/bombshelter.rb:79]: This should be a frozen constant

**Comment by DarthSim** [comment]
Splendid! Thanks for your contribution.

**Comment by DarthSim** [comment]
Released in v0.2.2

File: lib/carrierwave/bombshelter.rb [hunk 5]
```diff
@@ -13,6 +13,11 @@ module CarrierWave
   module BombShelter
     extend ActiveSupport::Concern
 
+    FILE_WRAPPERS = [
+      CarrierWave::SanitizedFile,
+      CarrierWave::Uploader::Base
+    ]
```
Comment (PikachuEXE) [inline lib/carrierwave/bombshelter.rb:19]: should freeze?

Comment (DarthSim) [inline lib/carrierwave/bombshelter.rb:19]: Don't think we'll win much by freezing it, so it's not necessarily.

**Pull request merged**